sys/decl.go

```diff
@@ -168,10 +183,13 @@ func (t *VmaType) Align() uintptr {
 
 type LenType struct {
 	TypeCommon
-	TypeSize  uintptr
-	BigEndian bool
-	ByteSize  uintptr // want size in multiple of bytes instead of array size
-	Buf       string
+	TypeSize    uintptr
```

Comment:
We now have 5 fields + 5 methods duplicated across 5 types.
It does not look good.
I think we should be able to do something along the following lines:

```go
type IntTypeCommon struct {
	TypeCommon
	TypeSize    uintptr
	BigEndian   bool
	BitfieldOff uintptr
	BitfieldLen uintptr
	BitfieldLst bool
}

func (t *IntTypeCommon) Size() uintptr {
	return t.TypeSize
}

func (t *IntTypeCommon) Align() uintptr {
	return t.Size()
}

func (t *IntTypeCommon) BitfieldOffset() uintptr {
	return t.BitfieldOff
}

func (t *IntTypeCommon) BitfieldLength() uintptr {
	return t.BitfieldLen
}

func (t *IntTypeCommon) BitfieldLast() bool {
	return t.BitfieldLst
}
```

Then embed `IntTypeCommon` instead of `TypeCommon` into these 5 types, and remove these methods from these types.

sysgen can use a common `commonInt` function similar to `common` that will generate all fields of `IntTypeCommon`.